lib/date_range.rb

```diff
@@ -0,0 +1,56 @@
+require 'Date'
+
+module Hotel
+  class DateRange
+    attr_accessor :start_date, :end_date
+
+    def initialize(start_date, end_date)
+      if start_date.class == String
+        start_date = Date.parse(start_date)
+      end
+      @start_date = start_date
+
+      if end_date.class == String
+        end_date = Date.parse(end_date)
+      end
+      @end_date = end_date
+
+      if @end_date && (@end_date < @start_date)
+        raise ArgumentError, "End time cannot be before start time."
```

Comment:
It's helpful to include the bad arguments in the message of your `ArgumentError`s:

```suggestion
        raise ArgumentError, "End time (#{end_date}) cannot be before start time (#{start_date})."
```